### DISCUSSION TITLE
怎样可以根据数据来动态选择icon标签，how can dynamically select an icon html-tag according to data?

### DISCUSSION THREAD

**Discussion by michaelxuzhi** [element-plus/element-plus#7178] [Q&A] [answered]
之前的图标选择，可以修改html标签的className来决定使用哪个icon，例如：`<i :class="item.icon"></i>`
现在的icon选择是以html标签的形式存在，无法根据传入的数据来动态选择。请问有什么办法吗？

selecting an icon in old version can be like this, change the 'className' of  a html-tag according to 'v-bind' instruction.
eg.  <i :class="item.icon"></i>
now the updated version render an icon by a html-tag, it doesn't allow me to change the html-tag by data, so it is not convenient and is not dynamic.
Is there any other method i can try like the previous way?

![icon](https://user-images.githubusercontent.com/16167158/163573203-cb85fb15-e5a4-4005-a15e-0e93ce8354ad.png)

**Comment by tolking** [comment, marked as answer]
[component is](https://github.com/element-plus/element-plus/blob/dev/docs/.vitepress/vitepress/components/globals/icons.vue#L60)

**Reply by michaelxuzhi** [reply]
非常感谢，已解决！thank u very much, it helps a lot !

**Comment by WestFarmer** [comment]
@tolking 
I am doing this, why it's not working?
```
<el-table-column fixed="right" width="50" align="center">
        <template #default="scope">
          <el-button v-for="(action, idx) in table.actions.filter((a) => a.type === 'row')" type="primary" size="small"
            link @click="fireAction"> <template #icon>
                  <component :is="`i-${action.icon}`" />
                </template> {{ action.text }}</el-button>
        </template>
      </el-table-column>
```

**Reply by michaelxuzhi** [reply]
it may not need "i-"
